### PR TITLE
UW-557 DRY out MPAS drivers

### PR DESCRIPTION
**Pull request by maddenp-noaa**
**Synopsis**

Factor out common code to a new `MPASBase` class. Functionality is otherwise unchanged.

**Type**

- [x] Code maintenance (refactoring, etc. without behavior change)

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/drivers/mpas.py [hunk 1]
```diff
@@ -166,10 +78,9 @@ def _driver_name(self) -> str:
...
         return STR.mpas
 
-    def _taskname(self, suffix: str) -> str:
+    @property
+    def _streams_fn(self) -> str:
         """
-        Returns a common tag for graph-task log messages.
-
-        :param suffix: Log-string suffix.
+        Teh streams filename.
```
Comment (NaureenBharwaniNOAA) [inline src/uwtools/drivers/mpas.py:84]:
```suggestion
        The streams filename.
```

Comment (maddenp-noaa) [reply src/uwtools/drivers/mpas.py]: Tahnks! :)

File: src/uwtools/drivers/mpas_init.py [hunk 2]
```diff
@@ -170,10 +82,9 @@ def _driver_name(self) -> str:
...
         return STR.mpasinit
 
-    def _taskname(self, suffix: str) -> str:
+    @property
+    def _streams_fn(self) -> str:
         """
-        Returns a common tag for graph-task log messages.
-
-        :param suffix: Log-string suffix.
+        Teh streams filename.
```
Comment (NaureenBharwaniNOAA) [inline src/uwtools/drivers/mpas_init.py:88]:
```suggestion
        The streams filename.
```

**Review by NaureenBharwaniNOAA — APPROVED** [review]
LGTM!

**Review by elcarpenterNOAA — APPROVED** [review]
Nice!

**Pull request merged**